src/app/admin/components/filter-search/filter-search.component.ts

```diff
@@ -77,34 +77,51 @@ export class FilterSearchComponent implements OnInit {
 
     //check for params in the query and add them to the filter dropdown bars
     const qParams = this.route.parent.snapshot.queryParams;
+
+    /**
+     * TODO: refactor this to send a complete search query
+     * at the first request instead of sending a request
+     * every time we emit an event on the toggle filter functions.
+     *
+     * this COULD be a cause of the client sending bad data back
+     * to the user, so just be aware.
+     */
+
     //if there are topics in the query, toggle them in the filter dropdown
-    if (qParams.topics) {
-      //more than one topic
+    if (qParams.topics && qParams.topics.length > 0) {
+      const topics = [];
       if (Array.isArray(qParams.topics)) {
         for (const topic of qParams.topics) {
-          this.toggleTopicFilter({ name: '', _id: topic });
+          topics.push({ name: '', _id: topic });
         }
-        // one topic
+
+        this.toggleTopicFilter(topics);
       } else {
-        this.toggleTopicFilter({ name: '', _id: qParams.topics });
+        this.toggleTopicFilter([{ name: '', _id: qParams.topics }]);
       }
     }
-    //if there are statuses in the query add them too the filter dropdowns
-    if (qParams.status) {
-      //multiple statuses in query
-      if (Array.isArray(qParams.status)) {
+
+    // if there are statuses in the query add them to the filter dropdowns
+    if (qParams.status && qParams.status.length > 0) {
```

Comment:
How come we have to check is qParams.status is an array on ln108 if we are checking `.length` on this line?

If status is a string then it is checking characters, if it array then its checking elements. I don't know if we need to check length in this case.

Reply:
removed this for converting to a map instead